### PR TITLE
binary

### PR DESCRIPTION
**Pull request by yingwang-us**

### CODE REVIEW

**Comment by alewitt2** [comment]
@yingwang-us have you had a chance to test this yet? You can do so by exporting the KUBECONFIG to your cluster you are testing against, then run `npm start | bunyan`

File: src/RemoteResourceS3DecryptController.js [hunk 1]
```diff
@@ -98,14 +98,20 @@ module.exports = class RemoteResourceS3DecryptController extends RemoteResourceS
 
     try {
       this.log.debug(`Downloaded from ${source}`);
+      const isCompressed = source.includes('.tar') || source.includes('.tgz');
```
Comment (adamkingit) [inline src/RemoteResourceS3DecryptController.js:111]: This isn't a definitive test to determine if the source is ascii armored or not. Try checking the mimetype on the response. Might have to fall back to just trying one, then checking for the appropriate exception, but that should be the last resort.

Comment (yingwang-us) [reply src/RemoteResourceS3DecryptController.js:111]: add isBinary flag based on res.headers['content-type'] === 'binary/octet-stream'. For normal text, COS returns application/octet-stream as the value. Tested against both binary and text rrs3d resources on minikube, download, pgp, uncompress work fine.

File: src/RemoteResourceS3DecryptController.js [hunk 2]
```diff
@@ -57,14 +57,24 @@ module.exports = class RemoteResourceS3DecryptController extends RemoteResourceS
...
+    reqOpt.resolveWithFullResponse = true;
     let res = await super.download(reqOpt);
     if (res.statusCode != 200) {
       return res;
     }
 
     let source = reqOpt.uri || reqOpt.url;
 
+    let isBinary = false;
+    if (res.headers['content-type'] === 'binary/octet-stream') {
```
Comment (adamkingit) [inline src/RemoteResourceS3DecryptController.js:72]: Have you confirmed we get binary/octet-stream for binay files?
What do we get for ascii armored files?

Comment (yingwang-us) [reply src/RemoteResourceS3DecryptController.js:72]: for ascii, we get `application/octet-stream`. for binary we will get `binary/octet-stream`

**Review by adamkingit — APPROVED** [review]

**Pull request merged**